src/test_report.py

```diff
@@ -133,49 +103,54 @@ def _create_test_report(self, root_node):
         )
         return content, subject
 
-    def _send_email(self, content, subject):
-        smtp = self._smtp_connect()
-        if smtp:
-            email_msg = self._create_email(
-                self._email_send_to, self._email_send_from,
-                subject, content
-            )
-            smtp.send_message(email_msg)
-            smtp.quit()
-
-    def _make_report(self, root_node, dump=True, send=False):
-        content, subject = self._create_test_report(root_node)
-        if dump:
-            print(content, flush=True)
-        if send:
-            self._send_email(content, subject)
+    def _send_report(self, subject, content):
+        try:
+            self._email_sender.create_and_send_email(
+                    subject, content
+                )
+        except Exception as err:
+            self._logger.log_message(logging.ERROR, err)
+
 
-    def run_loop(self):
+class TestReportLoop(TestReport):
+    """TestReport subclass to execute in a loop"""
+
+    def loop(self):
         sub_id = self._db.subscribe_node_channel(filters={
             'name': 'checkout',
             'state': 'done',
         })
 
-        self._logger.log_message(logging.INFO, "Listening for complete jobs")
+        self._logger.log_message(logging.INFO, "Listening for completed nodes")
         self._logger.log_message(logging.INFO, "Press Ctrl-C to stop.")
 
         try:
             while True:
                 root_node = self._db.receive_node(sub_id)
-                self._make_report(root_node, dump=True, send=True)
+                content, subject = self._get_report(root_node)
+                self._dump_report(content)
+                self._send_report(subject, content)
```

Comment:
```python
self._send_report(subject, content)
```
never raises an exception is it done on purpose?

Reply:
It is done to catch exceptions from `email_sender.create_and_send_email`. If I don't do that, traceback from `email_sender` will not be logged on the terminal.